spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -46,5 +52,24 @@ def test_project_explorer(project_explorer):
     assert project_explorer
 
 
+def test_copy_paste_files_or_paths(qtbot, tmpdir):
```

Comment:
I'd suggest using [pytest.mark.parametrize](https://docs.pytest.org/en/latest/parametrize.html) for this test to simplify it and eliminate duplicate code, while testing more edge cases. In particular, I'd suggest something like

```python
@pytest.mark.parametrize("path_mode", ["absolute", "relative"])
@pytest.mark.parametrize("file_paths", [["script.py"], ["script.py", "script1.py", "testdir/script2.py"], ["", "testdir"]])
```

which will test a single file, multiple files in different directories, and directories themselves (the working dir and a test dir) in both absolute and relative path mode. You'll need to modify the test function to take ``path_mode`` and ``file_paths`` as arguments and handle ``file_paths`` as lists, but it should result in a much simpler, cleaner and more comprehensive test.

Reply:
This comment is still valid, so I'll copy it over to my new review so it doesn't get lost.

Reply:
I Think this is resolved.